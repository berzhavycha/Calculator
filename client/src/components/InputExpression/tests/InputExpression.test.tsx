import { render } from '@testing-library/react';
import { InputExpression } from '../index';
import {vi} from 'vitest'

describe('InputExpression component', () => {
    test('matches the snapshot', () => {
        const mockProps = {
            expression: '2 + 2',
            inputName: 'expression-input',
            onExpressionChange: vi.fn(),
            onEnter: vi.fn(),
        };

        const { container } = render(
            <InputExpression
                expression={mockProps.expression}
                inputName={mockProps.inputName}
                onExpressionChange={mockProps.onExpressionChange}
                onEnter={mockProps.onEnter}
            />
        );

        expect(container).toMatchSnapshot();
    });
})