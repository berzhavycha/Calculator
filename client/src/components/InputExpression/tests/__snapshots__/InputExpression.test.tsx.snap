// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InputExpression component > matches the snapshot 1`] = `
<div>
  <input
    aria-label="expression-input"
    class="w-full p-2 border rounded outline-none text-right"
    type="text"
    value="2 + 2"
  />
</div>
`;
